guides/assets/esa-initial-flow.txt

```diff
@@ -0,0 +1,40 @@
+sequenceDiagram
+    participant User
+    participant App
+    participant Application Route
+    participant Authenticated Route
+    participant Unauthenticated Route
+    participant Session Service
+    participant Torii Provider
+    participant GitHub
+    participant Token Exchanger
+    User->>App: Use https://git-tool.com
+    App->>Application Route: transitionTo(application)
+    Application Route->>Application Route: subscribeToSessionEvents()
+    App->>Authenticated Route: transitionTo(index)
+    Authenticated Route->>Authenticated Route: beforeModel()
+    Authenticated Route->>Session Service: isAuthenticated?
+    Session Service-->>Authenticated Route: false
+    Authenticated Route->>Unauthenticated Route: transitionTo(authenticationRoute)
+    Unauthenticated Route->>Unauthenticated Route: beforeModel()
+    Unauthenticated Route->>Session Service: isAuthenticated?
+    Session Service-->>Unauthenticated Route: false
+    Unauthenticated Route->>Unauthenticated Route: render(loginTemplate)
```

Comment:
What if this transition went back to User instead? It definitely isn't ESA, but it ties the Ember behavior to what the user sees in a more concrete way, although with a little hand-waving.